Angular2-Starter-Project.md

```diff
@@ -0,0 +1,229 @@
+        h1 { color: darkgreen; font-family: Arial, Helvetica, sans-serif;}
+      `]
+    })
+    export class AppComponent { }
+```
+Every Angular 2 application has at least one component which is usually named **AppComponent**. A component manages a portion of the page using the template within it.
+
+Our component demonstrates the basic structure of any component you will ever write. It contains:
+* **import statements** to pull in other components we will need
+* **component decorator** which lets angular know which template to use and how the component will be created
+* **component class** which controls the components appearance and behavior
+
+Let's take a closer look at each line in our component.
+
+**Import**
+```
+import { Component } from '@angular/core';
+```
+Since angular applications are modular, we can import any other modules or libraries we may need. Here, we are importing the Angular 2 core module to give our component access to the @Component decorator. Every app will need this import to get started.
+
+Since we imported the ***Component*** function above, we can now use it to associate metadata with our component class that will tell Angular how our component should be created and the actions it will provide.
+
+**Component decorator**
+```
+    @Component({
+      selector: 'my-app',
+      template: '<h1>Free Code Camp Rocks!</h1>'
+    })
+```    
+Our metadata object has selector and template fields.
+* The ***selector*** specifies a CSS selector that indicates which HTML element will represent this component. The element we will use will be named "my-app". Angular will use this to create an instance of our component where it finds this element.
+* The ***template*** tells angular what template it will use for this component. This can refer to other Components or just a form of HTML that tells how to display the view for our component. Our template is just displaying an h1 element containing the text "Free Code Camp Rocks!".
+
+**Component Class**
+Our final line provides an empty class named ***AppComponent***
+```
+export class AppComponent { }
+```
+If we want to build a more complex component we can add logic and properties to the class. This component is extremely basic and simply displays some html, so its class will remain empty.
+
+We export our class so we can use it in other modules in our application.
+
+Next, we need to connect our root component to Angular. To do this we create another file in our app folder called main.ts that will have the following code:
+```
+    import { bootstrap }    from '@angular/platform-browser-dynamic';
+    import { AppComponent } from './app.component';
+    bootstrap(AppComponent);
+```
+This file imports the two items we need to start our app.
+* ***bootstrap*** - Angulars built in function that connects to the browser 
+* ***AppComponent*** - Our component we created above (which is why we exported it)
+We then call **bootstrap** function with our AppComponent
```

Comment:
- ~~***bootstrap***~~ **`bootstrap`** - Angulars built in ~~function~~ method that connects to the browser
- ~~***AppComponent***~~ **`AppComponent`** - Our component we created above (which is why we exported it) 

We then call ~~**bootstrap**~~ `bootstrap` ~~function~~ method with ~~our~~ ~~AppComponent~~`AppComponent`

```md
- **`bootstrap`** - Angulars built in method that connects to the browser
- **`AppComponent`** - Our component we created above (which is why we exported it) 

We then call `bootstrap` method with `AppComponent`
```